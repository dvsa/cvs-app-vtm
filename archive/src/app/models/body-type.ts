export interface BodyType {
  code: string;
  description: string;
}

export interface IBody {
  make: string;
  model: string;
  functionCode: string;
  conversionRefNo: string;
  bodyType: BodyType;
}

export interface BodyPSV extends Body {
  chassisMake: string;
  chassisModel: string;
}
