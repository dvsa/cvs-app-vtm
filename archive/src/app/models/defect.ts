interface Location {
  axleNumber?: number;
  horizontal?: string;
  vertical: string;
  longitudinal?: string;
  rowNumber: number;
  lateral: string;
  seatNumber: number;
}

interface AdditionalInformation {
  location: Location;
  notes: string;
}

export interface Defect {
  prohibitionIssued: boolean;
  additionalInformation: AdditionalInformation;
  itemNumber: number;
  deficiencyRef: string;
  stdForProhibition: boolean;
  deficiencyId: string;
  imNumber: number;
  deficiencyCategory: string;
  deficiencyText: string;
  prs: boolean;
  deficiencySubId?: any;
  imDescription: string;
  itemDescription: string;
}
