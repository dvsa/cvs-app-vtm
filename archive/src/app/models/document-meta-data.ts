export interface DocumentMetaData {
  uuid: string;
  fileName: string;
  metaName: string;
}

export interface DocumentInfo {
  metaName: string;
  file?: File;
}
