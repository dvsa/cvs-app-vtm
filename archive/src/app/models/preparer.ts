export interface Preparer {
  preparerId: string;
  preparerName: string;
}
