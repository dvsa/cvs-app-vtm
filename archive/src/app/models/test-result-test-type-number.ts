import { TestResultModel } from '@app/models/test-result.model';

export interface TestResultTestTypeNumber {
  testResultUpdated: TestResultModel;
  testTypeNumber: string;
  testResultsUpdated: TestResultModel[];
}
