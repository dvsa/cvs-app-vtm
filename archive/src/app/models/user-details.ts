export interface UserDetails {
  msUser: string;
  msOid: string;
}
