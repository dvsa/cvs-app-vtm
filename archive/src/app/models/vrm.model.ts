export interface VrmModel {
  vrm: string;
  isPrimary: boolean;
}
