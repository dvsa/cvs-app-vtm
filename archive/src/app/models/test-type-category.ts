import { TestCode } from '@app/models/test-code';

export interface TestTypeCategory {
  id: string;
  linkedIds: string[];
  name: string;
  testTypeName: string;
  forVehicleType: string[];
  forVehicleSize: string[];
  forVehicleConfiguration: string[];
  forVehicleAxles: number[];
  forEuVehicleCategory: string[];
  forVehicleClass: string[];
  forVehicleSubclass: string[];
  forVehicleWheels: number[];
  testCodes?: TestCode[];
  nextTestTypesOrCategories?: TestTypeCategory[];
}
