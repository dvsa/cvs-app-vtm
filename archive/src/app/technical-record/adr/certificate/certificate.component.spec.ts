import { AdditionalNotes } from './../../../models/adr-details';
import { CUSTOM_ELEMENTS_SCHEMA, Component, Input } from '@angular/core';
import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { SharedModule } from '@app/shared';
import { TESTING_UTILS } from '@app/utils/testing.utils';
import { AdrDetails } from '@app/models/adr-details';
import { CertificateComponent } from './certificate.component';

const adrWithCertRequested = TESTING_UTILS.mockAdrDetails({
  additionalNotes: {
    guidanceNotes: ['New certificate requested']
  } as AdditionalNotes
}) as AdrDetails;

describe('CertificateComponent', () => {
  let component: TestCertificateComponent;
  let fixture: ComponentFixture<TestCertificateComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [SharedModule],
      declarations: [
        CertificateComponent,
        TestCertificateComponent,
        TestCertificateEditComponent
      ],
      schemas: [CUSTOM_ELEMENTS_SCHEMA]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(TestCertificateComponent);
    component = fixture.componentInstance;
    component.edit = false;
    component.adrDetails = TESTING_UTILS.mockAdrDetails();
  });

  it('should create view only with populated data', () => {
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should render the editable component if edit is true', () => {
    component.adrDetails = adrWithCertRequested;
    component.edit = true;
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });
});

@Component({
  selector: 'test-vtm-certificate',
  template: `
    <vtm-certificate [edit]="edit" [adrDetails]="adrDetails"></vtm-certificate>
  `
})
class TestCertificateComponent {
  edit: boolean;
  adrDetails: AdrDetails;
}

@Component({
  selector: 'vtm-certificate-edit',
  template: `
    <div>Value of Certificate Request is {{ hasCertificateRequested }}</div>
  `
})
class TestCertificateEditComponent {
  @Input() hasCertificateRequested: boolean;
}
