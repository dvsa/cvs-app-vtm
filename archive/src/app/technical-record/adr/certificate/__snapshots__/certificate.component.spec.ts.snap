// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CertificateComponent should create view only with populated data 1`] = `
<test-vtm-certificate
  adrDetails={[Function Object]}
  edit="false"
>
  <vtm-certificate
    ng-reflect-adr-details="[object Object]"
    ng-reflect-edit="false"
  >
    
    
    <h2
      class="govuk-heading-m"
    >
      Certificate
    </h2>
    <dl
      class="govuk-summary-list"
    >
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key"
        >
          Required
        </dt>
        <dd
          class="govuk-summary-list__value"
          id="test-guidanceNotes"
        >
          
          <span>
            -
          </span>
        </dd>
        <dd
          class="govuk-summary-list__value"
        />
      </div>
    </dl>
    
    
  </vtm-certificate>
</test-vtm-certificate>
`;

exports[`CertificateComponent should render the editable component if edit is true 1`] = `
<test-vtm-certificate
  adrDetails={[Function Object]}
  edit={[Function Boolean]}
>
  <vtm-certificate
    ng-reflect-adr-details="[object Object]"
    ng-reflect-edit="true"
  >
    
    
    
    
    <vtm-certificate-edit
      ng-reflect-has-certificate-requested="true"
    >
      <div>
        Value of Certificate Request is true
      </div>
    </vtm-certificate-edit>
  </vtm-certificate>
</test-vtm-certificate>
`;
